server/controllers/student.py

```diff
@@ -120,40 +122,78 @@ def submit_assignment(name):
     group = Group.lookup(current_user, assign)
     user_ids = assign.active_user_ids(current_user.id)
     fs = assign.final_submission(user_ids)
-
     if not assign.uploads_enabled:
         flash("This assignment cannot be submitted online", 'warning')
         return redirect(url_for('.assignment', name=assign.name))
     if not assign.active:
         flash("It's too late to submit this assignment", 'warning')
         return redirect(url_for('.assignment', name=assign.name))
 
-    form = UploadSubmissionForm()
-    if form.validate_on_submit():
+
+    if request.method == "POST":
         backup = Backup(
             submitter=current_user,
             assignment=assign,
             submit=True,
         )
-        if form.upload_files.upload_backup_files(backup):
-            db.session.add(backup)
-            db.session.commit()
-            if assign.autograding_key:
-                try:
-                    submit_continous(backup)
-                except ValueError as e:
-                    logger.warning('Web submission did not autograde', exc_info=True)
-                    flash('Did not send to autograder: {}'.format(e), 'warning')
-            flash('Uploaded submission', 'success')
-            return redirect(url_for(
-                '.code',
-                name=assign.name,
-                submit=backup.submit,
-                bid=backup.id,
-            ))
+        assignment = backup.assignment
+        templates = assignment.files or []
+
+        files, binary_files = {}, {}
+
+        sorted_uploads = sorted(list(request.files.items()), key=lambda x: int(x[0][x[0].find('[')+1:-1]))
```

Comment:
This should be more parseable. 

It exists because `request.files.getlist` doesn't work with the way dropzone names it's files.